include/curve.h

```diff
@@ -152,6 +152,9 @@ class Curve {
   * parameter from to parameter to when robot is a box with side 2*radius
   */
   virtual bool intersects(const AlignedBox& box, T from, T to, T radius) const {
+    return false;
+    /*
+    DISABLED FOR NOW TODO:FIX
```

Comment:
Let's fix this, what is the exact problem?

Reply:
Causes an assertation error from Eigen3 such as 
`In file included from /usr/include/eigen3/Eigen/Core:347:0,
                 from /usr/include/eigen3/Eigen/Dense:1,
                 from /home/alpcevikel/splx/include/curve.h:3,
                 from /home/alpcevikel/splx/include/bezier.h:4,
                 from /home/alpcevikel/splx/test/catch_bezier.cpp:1:
/usr/include/eigen3/Eigen/src/Core/Matrix.h: In instantiation of ‘Eigen::Matrix<_Scalar, _Rows, _Cols, _Options, _MaxRows, _MaxCols>::Matrix(const Scalar&, const Scalar&, const Scalar&) [with _Scalar = double; int _Rows = 2; int _Cols = 1; int _Options = 0; int _MaxRows = 2; int _MaxCols = 1; Eigen::Matrix<_Scalar, _Rows, _Cols, _Options, _MaxRows, _MaxCols>::Scalar = double]’:
/home/alpcevikel/splx/include/curve.h:157:21:   required from ‘bool splx::Curve<T, DIM>::intersects(const AlignedBox&, T, T, T) const [with T = double; unsigned int DIM = 2; splx::Curve<T, DIM>::AlignedBox = Eigen::AlignedBox<double, 2>]’
/home/alpcevikel/splx/test/catch_bezier.cpp:97:1:   required from here
/usr/include/eigen3/Eigen/src/Core/util/StaticAssert.h:32:40: error: static assertion failed: THIS_METHOD_IS_ONLY_FOR_VECTORS_OF_A_SPECIFIC_SIZE
     #define EIGEN_STATIC_ASSERT(X,MSG) static_assert(X,#MSG);
                                        ^
/usr/include/eigen3/Eigen/src/Core/util/StaticAssert.h:153:3: note: in expansion of macro ‘EIGEN_STATIC_ASSERT’
   EIGEN_STATIC_ASSERT(TYPE::IsVectorAtCompileTime && TYPE::SizeAtCompileTime==SIZE, \
   ^~~~~~~~~~~~~~~~~~~
/usr/include/eigen3/Eigen/src/Core/Matrix.h:350:7: note: in expansion of macro ‘EIGEN_STATIC_ASSERT_VECTOR_SPECIFIC_SIZE’
       EIGEN_STATIC_ASSERT_VECTOR_SPECIFIC_SIZE(Matrix, 3)
       ^
`

Reply:
See this line in bezier.h:
 const VectorDIM rad(radius, radius, radius);

Here we have the assumption that we are working in 3d. We should generalize this to _DIM_ dimensions.

Reply:
I see your point. We are trying to initialize a n-dim vector with 3 points. It might be a solution to init the vector with default constructor and then assign the values in a for loop

Reply:
I looked for a more elegant solution but there is none I guess. lets do that.